### PR TITLE
Provide a new implementatio of `ct_find_uint8`

### PR DESCRIPTION
**Pull request by dinosaure**
This is a draft about a constant-time implementation of `ct_find_uint8` according @cfcs's proposal.

### CODE REVIEW

File: pk/rsa.ml [hunk 1]
```diff
@@ -338,8 +340,9 @@ module OAEP (H : Hash.S) = struct
   let eme_oaep_decode ?(label = Cstruct.empty) msg =
     let (b0, ms, mdb) = Cs.split3 msg 1 hlen in
     let db = MGF.mask ~seed:(MGF.mask ~seed:mdb ms) mdb in
-    let i  = Cs.ct_find_uint8 ~off:hlen ~f:((<>) 0x00) db |> Option.get ~def:0
-    in
+    let i  =
+      let v = Cs.ct_find_uint8 ~off:hlen ~f:((<>) 0x00) db in
+      if v = (-1) then 0 else v in
```
Comment (cfcs) [inline pk/rsa.ml]: `Eqaf.select` here

**Comment by cfcs** [comment]
@dinosaure isn't this a utility function that we should provide in `eqaf` ? It seems useful to other libraries too, and I think it would be better to maintain the constant-time things that are not crypto-related in `eqaf` instead of `mirage-crypto`

File: pk/rsa.ml (hunk 1 above)
Comment (dinosaure) [reply pk/rsa.ml]: How you want to use `select_int` here? I mean the condition here is `(-1)` and `choose_b` according to the behavior of `select_int` must be `0` or something else.

**Comment by dinosaure** [comment]
> @dinosaure isn't this a utility function that we should provide in eqaf ? It seems useful to other libraries too, and I think it would be better to maintain the constant-time things that are not crypto-related in eqaf instead of mirage-crypto

I agree with you but I did a release of `eqaf` few days ago. May be you can have an iterative plan where `ct_find_uint8` still is into `mirage-crypto` for the next release of it and prepare `eqaf` in consequence?

**Comment by hannesm** [comment]
I'd prefer that function to be in eqaf, tbh. once we have it in there (and released), we can use it in mirage-crypto. thanks.

**Comment by dinosaure** [comment]
> I'd prefer that function to be in eqaf, tbh. once we have it in there (and released), we can use it in mirage-crypto. thanks.

Ok,, I will post-pone that next week :+1:.

**Pull request closed without merging**